### PR TITLE
Adds credo linting

### PR DESCRIPTION
**Pull request by sobolevn**
Changes:
1. Adds new dependencies for testing and linting
2. All code is refactored to meet the linting rules
3. All functions changed from `function/0` to `function()/0`

Tests and dialixir are WIP (so, don't mind the unused dependencies at the moment)

### CODE REVIEW

File: lib/ueberauth/strategy/auth0/oauth.ex [hunk 1]
```diff
@@ -14,8 +14,10 @@ defmodule Ueberauth.Strategy.Auth0.OAuth do
         client_secret: {:system, "AUTH0_CLIENT_SECRET"}
   """
   use OAuth2.Strategy
+  alias OAuth2.Client
+  alias OAuth2.Strategy.AuthCode
 
-  def options() do
+  def options do
```
Comment (sntran) [inline lib/ueberauth/strategy/auth0/oauth.ex]: This is not a complain. I just dislike Credo's warning about this. A function, even with zero arity, should always have parentheses to indicate it is a function.

**Comment by sntran** [comment]
LGTM.

File: lib/ueberauth/strategy/auth0/oauth.ex (hunk 1 above)
Comment (sobolevn) [inline lib/ueberauth/strategy/auth0/oauth.ex]: Ok, you can create a custom config for `credo`, I would follow it.

Comment (sntran) [inline lib/ueberauth/strategy/auth0/oauth.ex]: You don't have to if you think `def options do` is the better way. I just voiced my opinion.

Comment (sobolevn) [inline lib/ueberauth/strategy/auth0/oauth.ex]: I am coming from strict `python`, so I think so too.

**Pull request merged**